### ISSUE TITLE
Wrong order when connecting points with line (frontend does not respect timestamps correctly)

### ISSUE THREAD

**Issue by brummbrum** [owntracks/frontend#67]
Using the frontend (docker version):
When showing location history (line) the points are connected in the order how they are listed in the .rec file stored by otrecorder rather than using the timestamp that the client reports together with each position. 
When clients loose cellular connectivity they cache positions and report them later. In otrecorder's .rec file these entries may be out of order, which is not a problem because clients report the positions together with correct timestamps.

frontend just connects the different entries "as a list" in the order how they appear in the .rec file. This occasionally creates nonsensical zig zag lines bouncing back and forth until the list follows a stringent chronological order again. Interestingly, otrecorder's built in linestring view does not suffer from this problem and connects all the points correctly in the order of the timestamps. It would be great if you could change the code in frontend to also respect the timestamps just like otrecorder linestring does. 

Simple example:

These 6 locations are listed in the same chronological order as the timestamps:

```
2021-11-01T20:03:30Z	*                 	{"_type":"location","acc":6,"alt":398,"batt":83,"bs":1,"conn":"m","lat":40.358487,"lon":-110.03499,"tid":"P","tst":1635797010,"vac":108,"vel":3}
2021-11-01T20:05:14Z	*                 	{"_type":"location","acc":6,"alt":335,"batt":83,"bs":1,"cog":68,"conn":"m","lat":40.358936,"lon":-110.034868,"tid":"P","tst":1635797114,"vac":3,"vel":5}
2021-11-01T20:05:42Z	*                 	{"_type":"location","acc":8,"alt":339,"batt":83,"bs":1,"cog":4,"conn":"m","lat":40.359393,"lon":-110.03482,"tid":"P","tst":1635797142,"vac":3,"vel":4}
2021-11-01T20:06:43Z	*                 	{"_type":"location","acc":8,"alt":345,"batt":83,"bs":1,"cog":351,"conn":"m","lat":40.359741,"lon":-110.035266,"tid":"P","tst":1635797203,"vac":3,"vel":5}
2021-11-01T20:07:11Z	*                 	{"_type":"location","acc":8,"alt":349,"batt":83,"bs":1,"cog":326,"conn":"m","lat":40.360183,"lon":-110.03544,"tid":"P","tst":1635797231,"vac":3,"vel":4}
2021-11-01T20:07:35Z	*                 	{"_type":"location","acc":8,"alt":355,"batt":83,"bs":1,"cog":331,"conn":"m","lat":40.360581,"lon":-110.035816,"tid":"P","tst":1635797255,"vac":3,"vel":5}
```

These are the same locations but the last position appears higher up in the list (which happens frequently when clients loose connectivity temporarily):

```
2021-11-01T20:03:30Z	*                 	{"_type":"location","acc":6,"alt":398,"batt":83,"bs":1,"conn":"m","lat":40.358487,"lon":-110.03499,"tid":"P","tst":1635797010,"vac":108,"vel":3}
2021-11-01T20:05:14Z	*                 	{"_type":"location","acc":6,"alt":335,"batt":83,"bs":1,"cog":68,"conn":"m","lat":40.358936,"lon":-110.034868,"tid":"P","tst":1635797114,"vac":3,"vel":5}
2021-11-01T20:05:42Z	*                 	{"_type":"location","acc":8,"alt":339,"batt":83,"bs":1,"cog":4,"conn":"m","lat":40.359393,"lon":-110.03482,"tid":"P","tst":1635797142,"vac":3,"vel":4}
2021-11-01T20:07:35Z	*                 	{"_type":"location","acc":8,"alt":355,"batt":83,"bs":1,"cog":331,"conn":"m","lat":40.360581,"lon":-110.035816,"tid":"P","tst":1635797255,"vac":3,"vel":5}
2021-11-01T20:06:43Z	*                 	{"_type":"location","acc":8,"alt":345,"batt":83,"bs":1,"cog":351,"conn":"m","lat":40.359741,"lon":-110.035266,"tid":"P","tst":1635797203,"vac":3,"vel":5}
2021-11-01T20:07:11Z	*                 	{"_type":"location","acc":8,"alt":349,"batt":83,"bs":1,"cog":326,"conn":"m","lat":40.360183,"lon":-110.03544,"tid":"P","tst":1635797231,"vac":3,"vel":4}

```

frontend renders the upper list correctly and the lower list wrongly with a zig zag.
otrecorder renders both lists correctly, i.e. the render appears exactly the same for both lists as it follows the timestamps.

**Comment by linusg** [comment]
Thanks for the bug report! I've seen this happen before, but I never investigated and just assumed that the entries were in chronological order but accuracy was low - which might not have been the case! The bug is fixed and I'll make a new release shortly. My only concern is how well the manual sorting performs on large datasets, but we'll see.

**Comment by linusg** [comment]
Released in [2.10.0](https://github.com/owntracks/frontend/releases/tag/v2.10.0), Docker image should be built shortly.

**Comment by brummbrum** [comment]
Wow! Thank you so much for fixing this so quickly!! Tested it immediately and it seems to work really well now even with some very messed up .rec files (I frequently run into these situations when connectivity of the clients gets lost in the backcountry). Actually I am surprised how well the manual sorting works and how elegantly you were able to fit it in. I have no clue about js and the .sort method and was a bit worried how it would treat entries without timestamps but it seems to behave gracefully. otrecorder required a little bit of code around this, [https://github.com/owntracks/recorder/blob/master/listsort.c](https://github.com/owntracks/recorder/blob/master/listsort.c). I have not noticed a performance hit in my environment even with large data sets. It seems that in my setup (ot-recorder runs on a raspberry pi) the ot-recorder api is the bottleneck rather than the js sorting.
